packages/core/src/helpers/createNodeFromContent.ts

```diff
@@ -46,18 +47,55 @@ export function createNodeFromContent(
 
       return schema.nodeFromJSON(content)
     } catch (error) {
+
+      if (options.errorOnInvalidContent) {
+        throw new Error('[tiptap error]: Invalid JSON content', { cause: error as Error })
+      }
+
       console.warn('[tiptap warn]: Invalid content.', 'Passed value:', content, 'Error:', error)
 
       return createNodeFromContent('', schema, options)
     }
   }
 
   if (isTextContent) {
-    const parser = DOMParser.fromSchema(schema)
+    let schemaToUse = schema
+    let hasInvalidContent = false
+
+    // Only ever check for invalid content if we're supposed to throw an error
+    if (options.errorOnInvalidContent) {
+      schemaToUse = new Schema({
+        topNode: schema.spec.topNode,
+        marks: schema.spec.marks,
+        nodes: schema.spec.nodes.append({
+          __tiptap__private__unknown__catch__all__node: {
+            content: 'inline*',
+            group: 'block',
+            parseDOM: [
+              {
+                tag: '*',
+                getAttrs: () => {
+                  hasInvalidContent = true
+                  return null
+                },
+              },
+            ],
+          },
+        }),
+      })
+    }
```

Comment:
Ended up finding the simplest way to express the catch-all node using prosemirror constructs